velox/functions/lib/Re2Functions.cpp

```diff
@@ -1137,20 +1380,45 @@ class PatternStringIterator {
     } else {
       if (currentChar == '_') {
         charKind_ = CharKind::kSingleCharWildcard;
+        nextStart_ = currentStart_ + 1;
       } else if (currentChar == '%') {
         charKind_ = CharKind::kAnyCharsWildcard;
+        nextStart_ = currentStart_ + 1;
       } else {
         charKind_ = CharKind::kNormal;
+
+        // Unicode.
+        if (currentChar & 0x80) {
+          auto numBytes = getCurrentCharBytes(pattern_.data() + currentStart_);
+          nextStart_ = currentStart_ + numBytes;
+        } else {
+          nextStart_ = currentStart_ + 1;
+        }
       }
-      nextStart_ = currentStart_ + 1;
     }
 
     return true;
   }
 
-  // Start index of the current character.
-  size_t currentStart() const {
-    return currentStart_;
+  // Char at current cursor, since it can be a multibyte character, here we use
+  // a string_view to represent.
+  std::string_view current() const {
+    // Unicode.
+    char startChar = charAt(currentStart_);
+    if (startChar & 0x80) {
+      return std::string_view(
+          pattern_.data() + currentStart_, nextStart_ - currentStart_);
+    }
+
+    // The following are Ascii code path.
+
+    // Escaped.
+    if (startChar == escapeChar_) {
```

Comment:
Just to clarify, escapeChar_ cannot be a unicode multi-byte character. Is this so?

Reply:
Yes, only `_` and `%` are escappable.

Reply:
I might be confused, but escapeChar_ here is not _ or %. It is user-specified character that can be used to escape _ and %. Can this user-specified character be a multi-byte character?

Reply:
Hmm.. I misunderstand your meaning before. You are right, escapeChar_ can be multi-byte character:

```
presto:sf1> select 'h_llo' LIKE 'h你_llo' ESCAPE '你';
 _col0
-------
 true
(1 row)
```

Can we support multi-byte escape char in another PR? To keep this PR focused

Reply:
It would be fine to support this in a follow up. Let's update this PR description to explain this limitation, add a TODO to the code and add a CHECK that escapeChar_ is ASCII. Thanks.

Reply:
ASCII escape char limitation is not something introduced by this PR, there are many `std::optional<char> escapeChar` in the code, make me don't know where to add the TODO, how about open an issue to track it?

Reply:
Let's find out the top-level call site and add a check there. It is no good to silently produce wrong results (even though this has been like that for some time).

Reply:
There is already a check to make sure the escape char is a single-byte char: https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/Re2Functions.cpp#L1331 , I added a TODO there.

Reply:
Got it. Thanks. Perhaps, update error message well. Currently it says "single character", but it really means "single ASCII character".

Reply:
Updated the error message.